### ISSUE TITLE
Error:bad argument #2 to '?' (out of range)

### ISSUE THREAD

**Issue by leetiantian** [fstrub95/Autoencoders_cf#8]
When I ran the main.lua with the movieLens-1M dataset, I found this error:

tian@ubuntu:~/RecSys/Autoencoders_cf-master/src$  th main.lua -file ../data/movieLens/movieLens-1M.t7 -conf ../conf/conf.movieLens.1M.V.lua -save network.t7 -type V -meta 1 -gpu 1
Options:    
- save      : network.t7  
- meta      : 1  
- conf      : ../conf/conf.movieLens.1M.V.lua  
- type      : V  
- seed      : 0  
- gpu       : 1  
- file      : ../data/movieLens/movieLens-1M.t7  
  Load training configuration...  
  Load data...  
  Loading data from disk...  

Users loaded      : 6040  
Items loaded      : 3706    

No Train ratings  : 900444  
No Test  ratings  : 99765  
Training ratio    : 0.90025584652808  
Training density  : 0.040226695210559   

Loading cunn... 
Loading data to GPU...  
Unbias the data...  
Data was successfully preloaded...  
Training network  
Start training : layer1-1  
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> output](1): cfn.AppenderSparseOut
  (2): nnsparse.Densify
  (3): nn.Linear(6059 -> 700)
  (4): nn.Tanh
  (5): cfn.AppenderOut
  (6): nn.Linear(719 -> 6040)
  (7): nn.Tanh
}
 [==>................. 1/15 ....................]  ETA: 0ms | Step: 0ms  
1/15     RMSE : 0.87886050271202     MAE : 0.6911922255146  
 [=====>.............. 2/15 ....................]  ETA: 2m49s | Step: 13s32ms  
2/15     RMSE : 0.8640292013925  MAE : 0.67872228449103 
 [========>........... 3/15 ....................]  ETA: 2m46s | Step: 13s867ms  
3/15     RMSE : 0.86040181710449     MAE : 0.67528439184942 
 [===========>........ 4/15 ....................]  ETA: 2m29s | Step: 13s587ms  
4/15     RMSE : 0.85725648624418     MAE : 0.67209682425549 
 [==============>..... 5/15 ....................]  ETA: 2m14s | Step: 13s422ms  
5/15     RMSE : 0.85668497973913     MAE : 0.67134923812129 
 [=================>.. 6/15 ....................]  ETA: 1m59s | Step: 13s313ms  
6/15     RMSE : 0.85593148123042     MAE : 0.67071014948892 
 [==================== 7/15 ....................]  ETA: 1m45s | Step: 13s153ms  
7/15     RMSE : 0.85595290538902     MAE : 0.67056789745237 
 [==================== 8/15 ....................]  ETA: 1m31s | Step: 13s34ms  
8/15     RMSE : 0.85618785292541     MAE : 0.67076270309566 
 [==================== 9/15 >...................]  ETA: 1m17s | Step: 12s973ms  
9/15     RMSE : 0.8567184022722  MAE : 0.67121781221904 
 [==================== 10/15 ==>................]  ETA: 1m4s | Step: 12s919ms  
10/15    RMSE : 0.85662947686255     MAE : 0.67104939328611 
 [================================>.............]  ETA: 51s453ms | Step: 12s863m 11/15 
11/15    RMSE : 0.85688711693553     MAE : 0.67124610363236 
 [===================================>..........]  ETA: 38s444ms | Step: 12s814m 12/15 
12/15    RMSE : 0.85718501506811     MAE : 0.67160791197393 
 [======================================>.......]  ETA: 25s570ms | Step: 12s785m 13/15 
13/15    RMSE : 0.85783763973171     MAE : 0.67219019824855 
 [=========================================>....]  ETA: 12s859ms | Step: 12s859m 14/15 
14/15    RMSE : 0.85773414173041     MAE : 0.67202228224052 
 [==================== 15/15 ==================>]  Tot: 3m757ms | Step: 12s911ms

15/15    RMSE : 0.85794016290016     MAE : 0.67204905269599 
Start training : layer2-1  
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> output](1): cfn.AppenderOut
  (2): nn.Linear(719 -> 500)
  (3): nn.Tanh
  (4): cfn.AppenderOut
  (5): nn.Linear(519 -> 700)
  (6): nn.Tanh
}
 [========>........... 1/5 .....................]  ETA: 0ms | Step: 0ms  
/home/tian/torch/install/bin/lua: bad argument #2 to '?' (out of range)
stack traceback:
    [C]: in ?
    [C]: in function '__index'
    misc/AutoEncoderTrainer.lua:67: in function 'Train'
    misc/AutoEncoderTrainer.lua:266: in function 'Execute'
    misc/TrainNetwork.lua:150: in function 'TrainNetwork'
    main.lua:69: in main chunk
    [C]: in function 'dofile'
    ...tian/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
    [C]: in ?

**Comment by fstrub95** [comment]
That is weird, I have to reproduce the results by myself to see what's happening.
In the meantime, you can use the movieLens-10M config file. If i remember well, you man have a final RMSE of 0.840. It is not the state of the art, but it is already a good benchmark

**Comment by Ashjawik** [comment]
Hey I got the same error when trying to customize config to 2 layer network. Any news on that?
Thanks and Cheers

**Comment by fstrub95** [comment]
Look into it next week, yet your problem looks different. In this case, the deep autoencoder is not properly fed but the input size match.

**Comment by tkipf** [comment]
Hi Florian, I get exactly the same issue when I run:

`th main.lua  -file ../data/movieLens/movieLens-1M.t7 -conf .
./conf/conf.movieLens.1M.V.lua  -save network.t7 -type V -meta 1 -gpu 1`

(After processing the movieLens-1M dataset with:
`th data.lua  -ratings ../data/movieLens/ratings-1M.dat -metaItem ../data/movieLens/movies-1M.dat -out ../data/movieLens/movieLens-1M.t7 -fileType movieLens -ratio 0.9` )

**Comment by fstrub95** [comment]
It is (finally) fixed!
https://github.com/fstrub95/Autoencoders_cf/commit/8434d4f1aec80b2c67f1d43c8bacd559b59b9c72
